### PR TITLE
Add a comment for function WithPlugin

### PR DESCRIPTION
**Pull request by yuxiaobo96**
Signed-off-by: yuxiaobo <yuxiaobogo@163.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -329,10 +329,3 @@ func newHealthzHandler(config *kubeschedulerconfig.KubeSchedulerConfiguration, s
 	}
 	return pathRecorderMux
 }
-
-// WithPlugin creates an Option based on plugin name and factory.
-func WithPlugin(name string, factory framework.PluginFactory) Option {
-	return func(registry framework.Registry) error {
-		return registry.Register(name, factory)
-	}
-}
```
Comment (ahg-g) [inline cmd/kube-scheduler/app/server.go]: this will be used by out-of-tree plugins.

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -316,13 +316,6 @@ func PodToleratesNodeTaints(pod *v1.Pod, meta Metadata, nodeInfo *schedulernodei
 	})
 }
 
-// PodToleratesNodeNoExecuteTaints checks if a pod tolerations can tolerate the node's NoExecute taints.
```
Comment (ahg-g) [inline pkg/scheduler/algorithm/predicates/predicates.go]: cleaned up in https://github.com/kubernetes/kubernetes/pull/86979

File: cmd/kube-scheduler/app/server.go (hunk 1 above)
Comment (ahg-g) [reply cmd/kube-scheduler/app/server.go]: please add a comment to say that "This function is used register out-of-tree plugins"

Comment (yuxiaobo96) [reply cmd/kube-scheduler/app/server.go]: Thanks your advice.

Comment (yuxiaobo96) [reply cmd/kube-scheduler/app/server.go]: done

File: cmd/kube-scheduler/app/server.go [hunk 3]
```diff
@@ -330,6 +330,7 @@ func newHealthzHandler(config *kubeschedulerconfig.KubeSchedulerConfiguration, s
 	return pathRecorderMux
 }
 
+// This function is used register out-of-tree plugins.
```
Comment (ahg-g) [inline cmd/kube-scheduler/app/server.go]: "used to register..."

Comment (yuxiaobo96) [reply cmd/kube-scheduler/app/server.go]: done

**Pull request merged**